README.rst

```diff
@@ -64,29 +79,70 @@ instances in return, one for text fields and the other for file uploads:
     def wsgi(environ, start_response):
       if environ["REQUEST_METHOD"] == "POST":
         forms, files = parse_form_data(environ)
-        
-        title = forms["title"]    # string
-        upload = files["upload"]  # MultipartPart
+
+        title  = forms["title"]  # string
+        upload = files["upload"] # MultipartPart
         upload.save_as(...)
 
-The ``parse_form_data`` helper function internally uses ``MultipartParser``, a
-streaming parser that reads from a ``multipart/form-data`` encoded binary data
-stream and emits ``MultipartPart`` instances as soon as a part is fully parsed.
-This is most useful if you want to consume the individual parts as soon as they
-arrive, instead of waiting for the entire request to be parsed:
+Note that form fields that are too large to fit into memory will end up as
+``MultipartPart`` instances in the ``files`` dict instead. This is to protect
+your app from running out of memory or crashing. ``MultipartPart`` instances are
+buffered to temporary files on disk if they exceed a certain size. The default
+limits should be fine for most use cases, but can be configured if you need to.
+See ``MultipartParser`` for details.
+
+Flask, Bottle & Co
+^^^^^^^^^^^^^^^^^^
+
+Most WSGI web framerworks already have multipart functionality built in, but
+you may still get better throughput for large files (or better limits control)
+by switching parsers: 
+
+.. code-block:: python
+
+    forms, files = multipart.parse_form_data(flask.request.environ)
+
+Legacy CGI
+^^^^^^^^^^
+
+If you are in the unfortunate position to have to rely on CGI, but can't use
+``cgi.FieldStorage`` anymore, it's possible to build a minimal WSGI environment
+from a CGI environment and use ``parse_form_data`` instead. This is *not*
+recommended though.
```

Comment:
It might be worth saying why it's not recommended - this leaves it ambiguous whether it's because of the deficiencies of the CGI protocol, or because of some kind of limitation of `multipart` in relation to CGI.

Reply:
I reworded it to be less ambiguous. Not our job to tell people to move away from CGI. Most of them know the risk already but are stuck with some legacy system they can't change.